src/scanners.rs

```diff
@@ -456,6 +456,143 @@ pub fn calc_indent(text: &str, max: usize) -> (usize, usize) {
     (i, spaces)
 }
 
+pub struct RefDefScan<'a> {
+    pub(crate) bytecount: usize,
+    pub(crate) label: &'a str,
+    pub(crate) dest: &'a str,
+    pub(crate) title: Option<&'a str>
+}
+
+pub fn scan_refdef(data: &str) -> Option<RefDefScan> {
+    let mut i = scan_whitespace_no_nl(data);
+
+    // defs can only be indented up to three spaces
+    if i > 3 {
+        return None;
+    }
+
+    // scan label
+    if scan_ch(&data[i..], b'[') == 0 {
+        return None;
+    }
+    i += 1;
+    let label_length = scan_while(&data[i..], |c| c != b']');
```

Comment:
This doesn't seem to be taking into account bracket nesting. I think you'll have to write a little state machine, it'll probably need to do backslash processing too.